src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -1720,19 +1798,56 @@ ReportDebugIpStatus::getPropertyTree20202( const xrt_core::device * _pDevice,
 
 
 void 
-ReportDebugIpStatus::writeReport( const xrt_core::device* _pDevice,
-                                  const boost::property_tree::ptree& /*_pt*/, 
+ReportDebugIpStatus::writeReport( const xrt_core::device* /*_pDevice*/,
+                                  const boost::property_tree::ptree& _pt, 
                                   const std::vector<std::string>& _elementsFilter,
                                   std::ostream & _output) const
 {
-  auto handle = _pDevice->get_device_handle();
 
-  _output << "Debug IP Status" << std::endl ;
+  const boost::property_tree::ptree& dbgIpStatus_pt = _pt.get_child("debug_ip_status");
+
+  //Print Overview
+  reportOverview(_output, dbgIpStatus_pt);
 
-  DebugIpStatusCollector collector(handle, false, _output);
-  collector.printOverview(_output);
-  collector.collect(_elementsFilter);
-  collector.printAllResults(_output);
+  // Process Element Filter
+  bool debugIpOpt[DEBUG_IP_TYPE_MAX];
+  // By default, enable status collection for all Debug IP types
+  std::fill(debugIpOpt, debugIpOpt + DEBUG_IP_TYPE_MAX, true);
+
+  bool filter = false;
+  if(_elementsFilter.size()) {
+    filter = true;
+    processElementFilter(debugIpOpt, _elementsFilter);
+  }
+
+  // Results
+  if (true == debugIpOpt[AXI_MM_MONITOR]) {
```

Comment:
Future consideration (doesn't need to be apart of this PR).

Instead of creating a collection of boolean values of debugIPOpt, you instead created a collection of callback functions, this code would be cleaner.